src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -320,7 +335,7 @@ describe('HeatmapPlot', () => {
 
     // On hovering over heatmap tracks
     await act(async () => {
-      vitesscePropsSpy.setTrackHighlight(['4', 0, 1, 2]);
+      await vitesscePropsSpy.setTrackHighlight(['4', 0, 1, 2]);
```

Comment:
why adding all these awaits?

Reply:
so that the async functions will resolve before moving on to the next line. If we don't do this, the test will continue without waiting for the resolution of the async calls. This happens a lot in test cases, and this is the source of the `UnhandledPromiseRejectionWarning` logs when we run tests. Sometimes, when we correctly wait for the promise to resolve, test fails because there are errors thrown when we resolve the promise chain.

Reply:
makes sense!